web/client/components/TOC/fragments/settings/VisibilityLimitsForm.jsx

```diff
@@ -0,0 +1,405 @@
+            const resolution = value / dpu;
+            return {
+                resolution: resolution,
+                scale: value,
+                zoom: getZoomFromResolution(resolution, resolutions)
+            };
+        }
+        return {
+            resolution: value,
+            scale: value * dpu,
+            zoom: getZoomFromResolution(value, resolutions)
+        };
+    }
+
+    const [options, setOptions] = useState([]);
+    const [maxValue, setMaxValue] = useState(maxResolution && {
+        resolution: maxResolution,
+        scale: maxResolution * dpu,
+        zoom: getZoomFromResolution(maxResolution, resolutions)
+    });
+    const [minValue, setMinValue] = useState(minResolution && {
+        resolution: minResolution,
+        scale: minResolution * dpu,
+        zoom: getZoomFromResolution(minResolution, resolutions)
+    });
+    const [loading, setLoading] = useState(false);
+    const [message, setMessage] = useState();
+    const [error, setError] = useState();
+    const isMounted = useRef();
+
+    useEffect(() => {
+        isMounted.current = true;
+        return () => {
+            isMounted.current = false;
+        };
+    }, []);
+
+    function clearAlters() {
+        setError(undefined);
+        setMessage(undefined);
+    }
+
+    function handleGetCapabilities() {
+        if (layer.type === 'wms' && !loading) {
+            setLoading(true);
+            clearAlters();
+            getLayerCapabilities(layer)
+                .toPromise()
+                .then((capabilities) => {
+                    if (isMounted.current) {
+                        const {
+                            maxScaleDenominator,
+                            minScaleDenominator
+                        } = capabilities;
+                        const newMaxValue = !isNil(maxScaleDenominator)
+                            && getResolutionObject(maxScaleDenominator, 'scale');
+                        const newMinValue = !isNil(minScaleDenominator)
+                            && getResolutionObject(minScaleDenominator, 'scale');
+                        const newMessageId = isNil(maxScaleDenominator) && isNil(minScaleDenominator)
+                            ? 'layerProperties.visibilityLimitsValuesUpdateUndefined'
+                            : 'layerProperties.visibilityLimitsValuesUpdate';
+                        setMessage(newMessageId);
+                        onChange({
+                            maxResolution: newMaxValue.resolution,
+                            minResolution: newMinValue.resolution
+                        });
+                        setLoading(false);
+                    }
+                })
+                .catch(() => {
+                    if (isMounted.current) {
+                        setLoading(false);
+                        setError('layerProperties.visibilityLimitsValuesError');
+                    }
+                });
+        }
+    }
+
+    useEffect(() => {
```

Comment:
this effect is executed when, limitsType or projection changes, but they are not used inside the effect.
maybe i'm wrong here, but can you clarify?

Reply:
After a double check those dependencies were not correct because the right ones should be dpu and resolutions. Going to fix this